SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityScheduler.java

```diff
@@ -384,6 +390,27 @@ public void drainPendingQueue() {
     );
   }
 
+  private void handlePendingRequestsForDeployKeySafe(
+    AtomicInteger obsoleteRequests,
+    AtomicInteger heldForScheduledActiveTask,
+    AtomicInteger totalNewScheduledTasks,
+    SingularityDeployKey deployKey,
+    List<SingularityPendingRequest> pendingRequestsForDeploy
+  ) {
+    try {
+      handlePendingRequestsForDeployKey(
+        obsoleteRequests,
+        heldForScheduledActiveTask,
+        totalNewScheduledTasks,
+        deployKey,
+        pendingRequestsForDeploy
+      );
+    } catch (Exception e) {
+      LOG.error("Error handling pending requests for {}, skipping", deployKey, e);
+      unscheduledTasksMetric.mark();
```

Comment:
Debated just reusing the lost tasks meter here, could be misleading (same task would get "lost" repeatedly) but would be fairly correct in spirit - @pschoenfelder you have any preference here?